src/vs/workbench/services/configurationResolver/browser/configurationResolverService.ts

```diff
@@ -164,12 +165,9 @@ export abstract class BaseConfigurationResolverService extends AbstractVariableR
 		}
 
 		// extract and dedupe all "input" and "command" variables and preserve their order in an array
-		const variables: string[] = [];
-		this.findVariables(configuration, variables);
-
 		const variableValues: IStringDictionary<string> = Object.create(null);
-
-		for (const variable of variables) {
+		const config = deepClone(configuration);
```

Comment:
Make a clone of the configuration so we do not manipulate the object passed into this function.